api/log.go

```diff
@@ -82,10 +93,30 @@ func GetBuildLogs(c *gin.Context) {
 		"user":  u.GetName(),
 	}).Infof("reading logs for build %s", entry)
 
+	// capture page query parameter if present
+	page, err := strconv.Atoi(c.DefaultQuery("page", "1"))
+	if err != nil {
+		//nolint:lll // ignore long line length due to error message
+		retErr := fmt.Errorf("unable to convert page query parameter for build %s: %w", entry, err)
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	// capture per_page query parameter if present
+	perPage, err := strconv.Atoi(c.DefaultQuery("per_page", "10"))
+	if err != nil {
+		retErr := fmt.Errorf("unable to convert per_page query parameter for build %s: %w", entry, err)
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	// ensure per_page isn't above or below allowed values
+	perPage = util.MaxInt(1, util.MinInt(500, perPage))
```

Comment:
I believe this should be `100` based off other API code that calls `util.MinInt(<int>, perPage)`

Reply:
I meant to have it do the same thing as before - the comment says default `100` but I accidentally used `10`. I'll fix that.

I think there can be more than 100 logs, especially if we add the InitStep logs from my proposal. So, I wanted to allow more than 100 per page, even though other endpoints allow at most 100 per page.

Do we really want the per_page max to be 100, the same as the default?

Reply:
Yeah, we shouldn't allow anything more than 100 results per page 👍

Especially for something like logs which can be quite taxing to retrieve depending on the size

Reply:
fwiw, i wasn't concerned about the 10 being the default, but you're right it maybe should have been marked as breaking.